### PR TITLE
feat: added default config file

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                     |
| Chore            | ✔                                                                       |

-->

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -12,13 +12,19 @@
...
+    "yamlFilePath": "/path/to/config/file.yaml",
     "cache": {
       "grids": ["epsg4326dir"],
       "request_format": "image/png",
       "upscale_tiles": 18,
       "type": "s3",
       "directory_layout": "tms"
+    },
+    "s3": {
+      "endpoint_url": "http://localhost:9000",
```
Comment (asafMasa) [inline config/default.json:26]: camelCase for all of the config file

File: src/common/interfaces.ts [hunk 2]
```diff
@@ -20,23 +20,37 @@ export interface OpenApiConfig {
...
 }
 
+export interface IMapProxyGlobalConfig {
+  cache: {
+    s3: {
+      endpoint_url: string;
+      bucket_name: string;
+    };
+  };
+}
```
Comment (asafMasa) [inline src/common/interfaces.ts:53]: press enter

File: src/initConfig.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import { extname } from 'path';
+import { container } from 'tsyringe';
+import { ServiceUnavailableError } from './common/exceptions/http/serviceUnavailableError';
+import { ILogger, IMapProxyConfig, IMapProxyJsonDocument } from './common/interfaces';
+import { convertJsonToYaml, convertYamlToJson } from './common/utils';
+import { Services } from './common/constants';
+
+// create default config yaml file if not exists
+export function initConfig(yamlFilePath: string, endPointUrl: string, bucket: string): void {
+  const mapproxyConfig: IMapProxyConfig = container.resolve(Services.MAPPROXY);
```
Comment (asafMasa) [inline src/initConfig.ts:11]: refactor mapproxyConfig

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/common/interfaces.ts (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/common/interfaces.ts:53]: fixed in 3d6efa8

File: config/default.json (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply config/default.json]: fixed in 6795f27

File: src/initConfig.ts (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/initConfig.ts]: fixed in eec1158

**Review by asafMasa — APPROVED** [review]

**Pull request merged**